packages/ui/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,78 @@
+import type { ButtonHTMLAttributes, PropsWithChildren } from 'react';
+import { forwardRef } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { Icon } from './Icon';
+import type { BaseProps } from './types';
+
+export interface ButtonProps extends BaseProps, ButtonHTMLAttributes<HTMLButtonElement> {
```

Comment:
Shouldn't we use namespace to exports props like we do in other projects? 🤔 

This is a preference thing, so would like to hear @e1himself's opinion on that

Reply:
I do prefer using namespaces for these: fewer symbols to export.

```ts
import { Button } from '__';

interface Props extends Button.Props {
  // ...
}
```

Reply:
I'm unable to export namespace with same name as a `const` variable

Reply:
Yeah, that's indeed unfortunate.

The best workaround we've found so far is to use `forwardRef` property instead of React's stock `ref`:

https://github.com/prezly/prezly/blob/39fc1fc99fffd5dbb51ec74a3a15adaa0b46c0a2/apps/backend/resources/javascripts/ui/Button/Button.tsx#L21

Reply:
thanks @e1himself , will update to use this approach

Reply:
Updated https://github.com/prezly/theme-kit-js/pull/404/commits/25d3d48a9248d6c224e985e7870e5f1681153016